src/rust/ide/src/model/project.rs

```diff
@@ -124,6 +124,10 @@ pub trait API:Debug {
     fn urm(&self) -> Rc<model::undo_redo::Manager>;
 }
 
+// Note: Needless lifetimes
+// ~~~~~~~~~~~~~~~~~~~~~~~~
+// See Note: [Needless lifetimes] is `model/project.rs`.
+
```

Comment:
Why there is such note? I don't see any "allow" needing explanation.

Reply:
This should have been placed in the execution context.